### PR TITLE
Reorg test

### PR DESCRIPTION
**Pull request by FreakyMaryk**
* Fixed old RPC test for reorg.

### CODE REVIEW

**Comment by nopslide** [comment]
This pull request **introduces 3 alerts** when merging 9aab34f2a432ec9ce3c5e16d97dd84fac4093e42 into 69bb948f6836f3d5c9772326880c390488534166 - [view on LGTM.com](https://lgtm.com/projects/g/zcoinofficial/zcoin/rev/pr-ba20f228d2824c50c4bf7d70a94e3bd0f1a8e891)

**new alerts:**

* 3 for &#39;import \*&#39; may pollute namespace

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by ultimaweapon** [comment]
You target `master` branch instead of `sigma`, is this intended?

**Comment by ultimaweapon** [comment]
Seem like it not run on CI yet so please fix enable it on CI first.

**Comment by nopslide** [comment]
This pull request **introduces 33 alerts** when merging fcf73642d2febac03e3ba551b96fb392563837a7 into c451cf5a97f4e06510568b31486c2104f6e1fafa - [view on LGTM.com](https://lgtm.com/projects/g/zcoinofficial/zcoin/rev/pr-ee1fe9e21d3df22e609535e09715786340d5c9cd)

**new alerts:**

* 26 for Duplicate key in dict literal
* 3 for &#39;import \*&#39; may pollute namespace
* 2 for Unused local variable
* 2 for Variable defined multiple times

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by nopslide** [comment]
This pull request **introduces 37 alerts** when merging 33eb2bf223f51f83b322bd597c217c6275ecf550 into c451cf5a97f4e06510568b31486c2104f6e1fafa - [view on LGTM.com](https://lgtm.com/projects/g/zcoinofficial/zcoin/rev/pr-6ec5a81acaebf0572094ee0ae5b4eff683d53bdf)

**new alerts:**

* 26 for Duplicate key in dict literal
* 5 for &#39;import \*&#39; may pollute namespace
* 4 for Unused local variable
* 2 for Variable defined multiple times

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by riordant** [comment]
Failure detected in `mempool_doublespend_oneblock.py`:
```
mempool_doublespend_oneblock.py:
Initializing test directory /tmp/test1sqreh4v/450
/home/tadhgriordan/Documents/master/zcoin/src/zcoind
Starting a process with: /home/tadhgriordan/Documents/master/zcoin/src/zcoind -datadir=/tmp/test1sqreh4v/450/node0 -server -keypool=1 -discover=0 -rest -reindex -dandelion=0 -mocktime=0 -checkmempool -debug=mempool
check double regular spend in multiple spend
Unexpected exception caught during testing: NameError("name 'try_create_tx_with_two_coinbase' is not defined",)
Stopping nodes
Not cleaning up dir /tmp/test1sqreh4v/450
Failed

stderr:
   File "/home/tadhgriordan/Documents/master/zcoin/qa/rpc-tests/test_framework/test_framework.py", line 149, in main
    self.run_test()
  File "/home/tadhgriordan/Documents/master/zcoin/qa/rpc-tests/mempool_doublespend_oneblock.py", line 40, in run_test
    spends2_raw = try_create_tx_with_two_coinbase(self.nodes[0], coinbase_txids1, coinbase_txids2,  node0_address, node0_address, 1, 1)

Pass: False, Duration: 6 s
```

**Comment by nopslide** [comment]
This pull request **introduces 35 alerts** when merging 76e01d926eef866556b306c4e0af73ba46ed0dfb into 015fd012582324a1a90054718f7b3b067c95de49 - [view on LGTM.com](https://lgtm.com/projects/g/zcoinofficial/zcoin/rev/pr-a7200de4c17cc1073f81b935bcd29ab4102b8bb7)

**new alerts:**

* 26 for Duplicate key in dict literal
* 5 for &#39;import \*&#39; may pollute namespace
* 2 for Unused local variable
* 2 for Variable defined multiple times

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by FreakyMaryk** [comment]
@riordant Now all should pass. Pls check again I commited changes.

**Comment by nopslide** [comment]
This pull request **introduces 35 alerts** when merging 810365b04af039172b9eb33a8d7452656a0cb58c into 8a33b1378a6c91033025307d6d4b2a1f202894b8 - [view on LGTM.com](https://lgtm.com/projects/g/zcoinofficial/zcoin/rev/pr-6b1f998159e4237babf07ea55824047ae6e63b11)

**new alerts:**

* 26 for Duplicate key in dict literal
* 5 for &#39;import \*&#39; may pollute namespace
* 2 for Unused local variable
* 2 for Variable defined multiple times

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: qa/rpc-tests/mempool_reorg.py [hunk 1]
```diff
@@ -67,8 +69,8 @@ def run_test(self):
         assert_raises(JSONRPCException, self.nodes[0].sendrawtransaction, timelock_tx)
 
         # Create 102_1 and 103_1:
-        spend_102_1_raw = create_tx(self.nodes[0], spend_102_id, node1_address, 49.98)
-        spend_103_1_raw = create_tx(self.nodes[0], spend_103_id, node1_address, 49.98)
+        spend_102_1_raw = create_tx(self.nodes[0], spend_102_id, node1_address, 0.1)
+        spend_103_1_raw = create_tx(self.nodes[0], spend_103_id, node1_address, 1)
```
Comment (thebevrishot) [inline qa/rpc-tests/mempool_reorg.py]: weird, Why these two lines pass? It should fail because input is not enough to pay the output and fee.

**Comment by nopslide** [comment]
This pull request **introduces 35 alerts** when merging 569aee84c3aff9af45049a08e99527e4e1f8afee into 8a33b1378a6c91033025307d6d4b2a1f202894b8 - [view on LGTM.com](https://lgtm.com/projects/g/zcoinofficial/zcoin/rev/pr-2963b842a9718ac6f9842dc034afcf0c1a72683a)

**new alerts:**

* 26 for Duplicate key in dict literal
* 5 for &#39;import \*&#39; may pollute namespace
* 2 for Unused local variable
* 2 for Variable defined multiple times

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by riordant** [comment]
Failure in `mempool_reorg`:
```
[1mmempool_reorg.py[0m:
Initializing test directory /tmp/testjjuf7pr0/571
/var/lib/jenkins/workspace/zcoin_main_PR-368/dist/src/zcoind
Starting a process with: /var/lib/jenkins/workspace/zcoin_main_PR-368/dist/src/zcoind -datadir=/tmp/testjjuf7pr0/571/node0 -server -keypool=1 -discover=0 -rest -reindex -dandelion=0 -mocktime=0 -checkmempool -debug=mempool
/var/lib/jenkins/workspace/zcoin_main_PR-368/dist/src/zcoind
Starting a process with: /var/lib/jenkins/workspace/zcoin_main_PR-368/dist/src/zcoind -datadir=/tmp/testjjuf7pr0/571/node1 -server -keypool=1 -discover=0 -rest -reindex -dandelion=0 -mocktime=0 -checkmempool -debug=mempool
Assertion failed: Block sync failed
Stopping nodes
Not cleaning up dir /tmp/testjjuf7pr0/571
Failed
```

**Comment by FreakyMaryk** [comment]
> Failure in `mempool_reorg`:
> 
> ```
> [1mmempool_reorg.py�[0m:
> Initializing test directory /tmp/testjjuf7pr0/571
> /var/lib/jenkins/workspace/zcoin_main_PR-368/dist/src/zcoind
> Starting a process with: /var/lib/jenkins/workspace/zcoin_main_PR-368/dist/src/zcoind -datadir=/tmp/testjjuf7pr0/571/node0 -server -keypool=1 -discover=0 -rest -reindex -dandelion=0 -mocktime=0 -checkmempool -debug=mempool
> /var/lib/jenkins/workspace/zcoin_main_PR-368/dist/src/zcoind
> Starting a process with: /var/lib/jenkins/workspace/zcoin_main_PR-368/dist/src/zcoind -datadir=/tmp/testjjuf7pr0/571/node1 -server -keypool=1 -discover=0 -rest -reindex -dandelion=0 -mocktime=0 -checkmempool -debug=mempool
> Assertion failed: Block sync failed
> Stopping nodes
> Not cleaning up dir /tmp/testjjuf7pr0/571
> Failed
> ```

It is happened from time to time. Need restart

**Pull request merged**